modules/ROOT/pages/provisioning-raspberry-pi4.adoc

```diff
@@ -0,0 +1,188 @@
+
+To run FCOS on a Raspberry Pi 4 via U-Boot the SD card or USB disk needs to be prepared on another system and then the disk moved to the RPi4. After writing FCOS to the disk a few more files will need to be copied in place on the EFI partition of the FCOS disk. Check out the https://www.raspberrypi.com/documentation/computers/configuration.html#boot-folder-contents[Raspberry Pi Documentation] to read more about what these files are for.
+
+In this case we can grab these files from the `uboot-images-armv8`, `bcm283x-firmware`, `bcm283x-overlays` RPMs from the Fedora Linux repositories. First download them and store them in a temporary directory on your system:
+
+```nohighlight
+RELEASE=34 # The target Fedora Release. Use the same one that current FCOS is based on.
+mkdir -p /tmp/RPi4boot/boot/efi/
+sudo dnf install -y --downloadonly --release=$RELEASE --forcearch=aarch64 --destdir=/tmp/RPi4boot/  uboot-images-armv8 bcm283x-firmware bcm283x-overlays
+```
+
+Now extract the contents of the RPMs and copy the proper `u-boot.bin` for the RPi4 into place:
+
+```nohighlight
+for rpm in /tmp/RPi4boot/*rpm; do rpm2cpio $rpm | sudo cpio -idv -D /tmp/RPi4boot/; done
+sudo mv /tmp/RPi4boot/usr/share/uboot/rpi_4/u-boot.bin /tmp/RPi4boot/boot/efi/rpi4-u-boot.bin
+```
+
+Run `coreos-installer` to install to the target disk. There are https://coreos.github.io/coreos-installer/getting-started/[various ways] to run `coreos-installer` and install to a target disk. We won't cover them all here, but this workflow most closely mirrors the xref:bare-metal.adoc#_installing_from_the_container["Installing from the container"] documentation.
+
+```nohighlight
+FCOSDISK=/dev/sdX
+sudo coreos-installer install --architecture=aarch64 -i config.ign $FCOSDISK
+```
+
+NOTE: Make sure you provide an xref:producing-ign.adoc[Ignition config] when you run `coreos-installer`.
+
+Now mount up the EFI partition and copy the files over:
+
+```nohighlight
+mkdir /tmp/FCOSEFIpart
+sudo mount ${FCOSDISK}2 /tmp/FCOSEFIpart
+sudo rsync -avh --ignore-existing /tmp/RPi4boot/boot/efi/ /tmp/FCOSEFIpart/
+sudo umount ${FCOSDISK}2
+```
+
+Now take the USB/SD card and attach it to the RPi4 and boot.
+
+TIP: It can take some time to boot, especially if the disk is slow. Be patient. You may not see anything on the screen for 20-30 seconds.
+
+
+## Installing FCOS and Booting via UEFI
+
+There is a UEFI firwmare implementation for the RPi4 (https://github.com/pftf/RPi4/[pftf/RPi4]) that attempts to make the RPi4 ServerReady (SBBC compliant) and allows you to pretend that the RPi4 is just like any other server hardware with UEFI.
+
+You can write the firwmare to a disk (USB or SD card) and then boot/install FCOS as you would on any bare metal server. However, the firmware files need to be on an SD card or USB disk and will take up either the SD card slot or a USB slot. Depending on your needs this may be acceptable or not. Depending on the answer you have a few options:
```

Comment:
```suggestion
You can write the firmware to a disk (USB or SD card) and then boot/install FCOS as you would on any bare metal server. However, the firmware files need to be on an SD card or USB disk and will take up either the SD card slot or a USB slot. Depending on your needs this may be acceptable or not. Depending on the answer you have a few options:
```